cdap-app-fabric/src/main/java/io/cdap/cdap/app/metrics/ProgramUserMetrics.java

```diff
@@ -42,4 +52,14 @@ public void count(String metricName, int delta) {
   public void gauge(String metricName, long value) {
     metricsContext.gauge(metricName, value);
   }
+
+  @Override
+  public Metrics child(Map<String, String> tags) {
+    return new ProgramUserMetrics(metricsContext.childContext(tags), false);
+  }
+
+  @Override
+  public Map<String, String> getTags() {
+    return metricsContext.getTags();
```

Comment:
is this an unmodifiable map? If not, we should make sure it is.

Reply:
Context keeps immutable map as member so returned map is unmodifiable - https://github.com/cdapio/cdap/blob/develop/cdap-watchdog/src/main/java/io/cdap/cdap/metrics/collect/AggregatedMetricsCollectionService.java#L226